### PR TITLE
python3Packages.name-that-hash: init at 1.10

### PR DESCRIPTION
**Pull request by eyJhb**
###### Things done
Build, tested both the binary as well as the binary.
Unsure if this is the correct way to get the binary named `nth`.

Also, had to patch `pyproject.toml`until a new version is released.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Related to https://github.com/NixOS/nixpkgs/issues/81418

### CODE REVIEW

File: pkgs/development/python-modules/name-that-hash/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, click
+, rich
+}:
+
+buildPythonPackage rec {
+  pname = "nth";
```
Comment (fabaff) [inline pkgs/development/python-modules/name-that-hash/default.nix:10]: The Python module should be named `name-that-hash`.

```suggestion
  pname = "name-that-hash";
```

Add it to `all-packages.nix` as `nth` with `toPythonApplication`.

File: pkgs/development/python-modules/name-that-hash/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchFromGitHub {
+    owner = "HashPals";
+    repo = "Name-That-Hash";
+    rev = version;
+    hash = "sha256-3sddUPoC3NfKQzmNgqPf/uHaYN9VZBqsmV712uz1Phg=";
+  };
+
+  # TODO remove on next update which bumps rich
+  postPatch = ''
+    sed -i 's|rich = ">=9.9,<11.0"|rich = ">=9.9"|g' pyproject.toml
```
Comment (fabaff) [inline pkgs/development/python-modules/name-that-hash/default.nix:23]: With `substituteInPlace` would do the same as `sed`.

File: pkgs/development/python-modules/name-that-hash/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  propagatedBuildInputs = [
+    click
+    rich
+  ];
+
+  meta = with lib; {
+    description = "Don't know what type of hash it is? Name That Hash will name that hash type! Identify MD5, SHA256 and 300+ other hashes.";
+    homepage = "https://github.com/HashPals/Name-That-Hash";
+    license = with licenses; [ gpl3Only ];
```
Comment (RasmusRendal) [inline pkgs/development/python-modules/name-that-hash/default.nix:38]: Seems that the license is actually gpl3Plus.

Comment (fabaff) [inline pkgs/development/python-modules/name-that-hash/default.nix:38]: License is [GPLv3+](https://github.com/HashPals/Name-That-Hash/blob/main/pyproject.toml#L6).

```suggestion
    license = with licenses; [ gpl3Plus ];
```

**Review by fabaff — COMMENTED** [review]
The upstream repo contains [tests](https://github.com/HashPals/Name-That-Hash/tree/main/tests). Could you check if they could be executed?

Add `pythonImportsCheck` as additional safeguard.

**Review by RasmusRendal — CHANGES_REQUESTED** [review]
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64 linux
- [x] executables tested on x86_64 linux
- [x] all depending packages build

File: pkgs/development/python-modules/name-that-hash/default.nix (hunk 3 above)
Comment (eyJhb) [reply pkgs/development/python-modules/name-that-hash/default.nix]: Fixed

**Comment by eyJhb** [comment]
> The upstream repo contains [tests](https://github.com/HashPals/Name-That-Hash/tree/main/tests). Could you check if they could be executed?
> 
> Add `pythonImportsCheck` as additional safeguard.

I have added it to check imports, should be OK now.
I am unsure if it runs the tests, however I couldn't find much on it.

I have fixed all the other comments now :)

**Review by RasmusRendal — APPROVED** [review]

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/172296#issuecomment-11247155896)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-172296]$ ./results/nth/bin/nth 
Usage:  [OPTIONS]

  Name That Hash - Instantly name the type of any hash!
[...]
```
Looks good to me.

**Pull request merged**